### PR TITLE
Fix memleaks in libr/util/table.c

### PR DESCRIPTION
**Pull request by pelijah**

### CODE REVIEW

File: libr/util/table.c [hunk 1]
```diff
@@ -363,7 +366,9 @@ R_API char *r_table_tostring(RTable *t) {
 			maxlen = R_MAX (maxlen, ll);
 		}
 		int len = r_str_len_utf8_ansi (r_strbuf_get (sb));
-		r_strbuf_appendf (sb, "\n%s\n", r_str_repeat (h_line, R_MAX (maxlen, len)));
+		char *l = r_str_repeat (h_line, R_MAX (maxlen, len));
+		r_strbuf_appendf (sb, "\n%s\n", l);
```
Comment (radare) [inline libr/util/table.c]: check for null

File: libr/util/str.c [hunk 2]
```diff
@@ -2959,7 +2959,7 @@ R_API const char *r_str_pad(const char ch, int sz) {
 	return pad;
 }
 
-R_API const char *r_str_repeat(const char *ch, int sz) {
+R_API char *r_str_repeat(const char *ch, int sz) {
 	RStrBuf *buf;
 	int i;
 	if (sz < 0) {
```
Comment (radare) [inline libr/util/str.c]: missing space below in the strdup("")

File: libr/util/str.c [hunk 3]
```diff
@@ -2959,7 +2959,7 @@ R_API const char *r_str_pad(const char ch, int sz) {
 	return pad;
 }
 
-R_API const char *r_str_repeat(const char *ch, int sz) {
+R_API char *r_str_repeat(const char *ch, int sz) {
 	RStrBuf *buf;
```
Comment (radare) [inline libr/util/str.c]: define in the same line that is assigned

**Comment by radare** [comment]
lgtm but the XX seems to be a crash in iS?

**Comment by pelijah** [comment]
Weird... even on master `iS` produces different output than https://github.com/radareorg/radare2-regressions/commit/b466ab9ef8921277a7be3602841f2b8efaf56ebc.
изображение

**Comment by radare** [comment]
You are not passing any query to iS

Try: iS size/sort,vaddr/cols/name

**Comment by pelijah** [comment]
Ah got it. Yep r2 crashes.

**Comment by pelijah** [comment]
Memleak fix introduces some sort of magic when `r_table_row_free()` calls itself
```
    #0 0x7fb41eb2f672 in r_list_purge /home/user/radare2/build_asan/../libr/util/list.c:69:24
    #1 0x7fb41eb2f94f in r_list_free /home/user/radare2/build_asan/../libr/util/list.c:82:3
    #2 0x7fb41eae0c64 in r_table_row_free /home/user/radare2/build_asan/../libr/util/table.c:58:2
    #3 0x7fb41eb2f8e3 in r_list_delete /home/user/radare2/build_asan/../libr/util/list.c:106:3
    #4 0x7fb41eb2f6e3 in r_list_purge /home/user/radare2/build_asan/../libr/util/list.c:72:3
    #5 0x7fb41eb2f94f in r_list_free /home/user/radare2/build_asan/../libr/util/list.c:82:3
    #6 0x7fb41eae0c64 in r_table_row_free /home/user/radare2/build_asan/../libr/util/table.c:58:2
    #7 0x7fb41eb2f8e3 in r_list_delete /home/user/radare2/build_asan/../libr/util/list.c:106:3
    #8 0x7fb41eb2f6e3 in r_list_purge /home/user/radare2/build_asan/../libr/util/list.c:72:3
    #9 0x7fb41eb2f94f in r_list_free /home/user/radare2/build_asan/../libr/util/list.c:82:3
    #10 0x7fb41eae0ea1 in r_table_free /home/user/radare2/build_asan/../libr/util/table.c:81:2
    #11 0x7fb41bf7f471 in bin_sections /home/user/radare2/build_asan/../libr/core/cbin.c:2789:2
```

**Comment by pelijah** [comment]
But there are another memleak in `r_table_columns()`

**Pull request merged**